centml/cli/cluster.py

```diff
@@ -158,82 +175,38 @@ def get(type, id):
-    func = click.option("--command", type=str, required=False, default=None, help="Define a command for a container")(
-        func
-    )
-    func = click.option("--command_args", multiple=True, type=str, default=None, help="List of command arguments")(func)
-    return func
-
+    return list(hw_to_id_map.keys())
 
-# Define compute specific options
-def compute_options(func):
-    func = click.option("--username", prompt="Username", type=str, help="Username")(func)
-    func = click.option("--password", prompt="Password", hide_input=True, type=str, help="password")(func)
-    func = click.option(
-        "--ssh_key", prompt="Add ssh key", default="", type=str, help="Would you like to add an SSH key?"
-    )(func)
-    return func
 
-
-# Main command group
 @click.group(help="Create a new deployment")
-@click.pass_context
-def create(ctx):
+def create():
     pass
 
 
-# Define the inference subcommand
 @create.command(name="inference", help="Create an inference deployment")
-@common_options
-@inference_options
-@click.pass_context
-def create_inference(ctx, **kwargs):
+@click.option("--name", "-n", prompt="Name", help="Name of the deployment")
+@click.option("--image", "-i", prompt="Image", help="Container image")
+@click.option("--hardware", "-h", prompt="Hardware", type=HardwarePricingTier(), help="Hardware instance type")
+@click.option("--port", "-p", prompt="Port", type=int, help="Port to expose")
+@click.option("--env", type=InferenceEnvType(), help="Environment variables in the format KEY=VALUE", multiple=True)
+@click.option("--min_replicas", default="1", prompt="Min replicas", type=click.IntRange(1, 10))
+@click.option("--max_replicas", default="1", prompt="Max replicas", type=click.IntRange(1, 10))
+@click.option("--health", default="/", prompt="Health check", help="Health check endpoint")
+@click.option("--is_private", default=False, type=bool, prompt="Is private?", help="Is private endpoint?")
+@click.option("--timeout", prompt="Max concurrency", default=0, type=int)
+@click.option("--command", type=str, required=False, default=None, help="Define a command for a container")
+@click.option("--command_args", multiple=True, type=str, default=None, help="List of command arguments")
+def create_inference(
+    name, image, hardware, port, env, min_replicas, max_replicas, health, is_private, timeout, command, command_args
+):
     click.echo("Creating inference deployment with the following options:")
 
-    name = kwargs.get("name")
-    image = kwargs.get("image")
-    port = kwargs.get("port")
-    is_private = kwargs.get("is_private")
-    hardware = kwargs.get("hardware")
-    health = kwargs.get("health")
-    min_replicas = kwargs.get("min_replicas")
-    max_replicas = kwargs.get("max_replicas")
-    env = kwargs.get("env")
-    command = kwargs.get("command")
-    command_args = kwargs.get("command_args")
-    timeout = kwargs.get("timeout")
-
     hw_to_id_map, _ = get_hw_to_id_map()
```

Comment:
is it possible to get this map from the HardwarePricingTier object? We have already made the API call to fetch the hardware instances. If we can reuse that without making another call, that would be great.

Reply:
done